src/OSPSuite.Core/Domain/Services/ValueComparer.cs

```diff
@@ -30,32 +22,50 @@ public static bool AreValuesEqual(double value1, double value2, double relTol)
       }
 
       /// <summary>
-      ///   returns true if the values of the given parameters are equals otherwise false
+      ///    returns <c>true</c> if the values of the given parameters are equals otherwise <c>false</c>
       /// </summary>
       public static bool AreValuesEqual(IParameter parameter1, IParameter parameter2)
       {
+         if (parameter1 == null && parameter2 == null)
+            return true;
+
+         if (parameter1 == null || parameter2 == null)
+            return false;
+
          return AreValuesEqual(parameter1.Value, parameter2.Value);
       }
 
       /// <summary>
-      ///   returns true if the percentile of the given parameters are equals otherwise false
+      ///    returns <c>true</c> if the value of <paramref name="parameter"/> is equal to <paramref name="value"/> otherwise <c>false</c>
       /// </summary>
-      public static bool ArePercentilesEqual(IDistributedParameter parameter1, IDistributedParameter parameter2)
+      public static bool AreValuesEqual(IParameter parameter, double value)
       {
-         return ArePercentilesEqual(parameter1.Percentile, parameter2.Percentile);
+         if (parameter == null)
+            return false;
+
+         return AreValuesEqual(parameter.Value, value);
       }
 
-      public static bool ArePercentilesEqual(double percentile1, double percentile2)
+      /// <summary>
+      ///    returns <c>true</c> if the percentile of the given parameters are equals otherwise <c>false</c>
+      /// </summary>
+      public static bool ArePercentilesEqual(IDistributedParameter parameter1, IDistributedParameter parameter2)
       {
-         return AreValuesEqual(percentile1, percentile2, 1e-2);
+         if (parameter1 == null && parameter2 == null)
+            return true;
+
+         if (parameter1 == null || parameter2 == null)
+            return false;
+
+         return ArePercentilesEqual(parameter1.Percentile, parameter2.Percentile);
       }
 
-      public static bool AreValuesEqual(float value1, float value2)
+      public static bool ArePercentilesEqual(double percentile1, double percentile2)
       {
-         return AreValuesEqual(value1, value2, Constants.FLOAT_RELATIVE_EPSILON);
+         return AreValuesEqual(percentile1, percentile2, Constants.DOUBLE_PERCENTILE_RELATIVE_TOLERANCE);
       }
 
-      public static bool AreValuesEqual(float value1, float value2, float relTol)
+      public static bool AreValuesEqual(float value1, float value2, float relTol = Constants.FLOAT_RELATIVE_EPSILON)
       {
          // two float NaN are not equal! Wonder of .NET
          if (float.IsNaN(value1))
```

Comment:
so if `value1 != NaN & value2 = NaN`, then `diff `will become `NaN `as well.
As long as  `NaN <= relTol` returns false (which is not obvious) (and having a test for that) it's OK